packages/grafana-ui/src/components/uPlot/Plot.tsx

```diff
@@ -83,12 +83,7 @@ export class UPlotChart extends Component<PlotProps, UPlotChartState> {
   componentDidUpdate(prevProps: PlotProps) {
     let { plot } = this.state;
 
-    if (!sameDims(prevProps, this.props)) {
-      plot?.setSize({
```

Comment:
@leeoniya shouldn't `plot.setSize` do the job? Looks like the config doesn't change, at least explicitly.

Reply:
I think I found the problem, we are checking data structure but not data itself. That's why is doesn't trigger config update.
https://github.com/grafana/grafana/blob/d4c28aa7574e32b8c5e01efd255b685096c2378f/packages/grafana-ui/src/components/Sparkline/Sparkline.tsx#L88-L90